driver-core/src/main/com/mongodb/internal/connection/MongoCredentialWithCache.java

```diff
@@ -63,15 +67,34 @@ public void putInCache(final Object key, final Object value) {
         cache.set(key, value);
     }
 
+    public OidcCacheEntry getOidcCacheEntry() {
+        return cache.oidcCacheEntry;
+    }
+
+    public void setOidcCacheEntry(final OidcCacheEntry oidcCacheEntry) {
+        this.cache.oidcCacheEntry = oidcCacheEntry;
+    }
```

Comment:
These two methods are now visible to all `Authenticator` implementations via `Authenticator.getMongoCredentialWithCache` even though they are specific to `OidcAuthenticator`. Note how the original `getFromCache`/`putInCache` methods are not specific to any implementation of the `Authenticator` interface, and represent API of a generic single-key-value cache.

It appears to me that `OidcAuthenticator` should do what other authenticators do: store what it needs in `MongoCredentialWithCache.Cache` using a key specific to `OidcAuthenticator. More specifically

1. introduce a key specific to `OidcAuthenticator`;
2. store/access `MutableOidcCacheEntry` using that key in `MongoCredentialWithCache.Cache`, where `MutableOidcCacheEntry` is

```java
private final class OidcAuthenticator.MutableOidcCacheEntry {
    volatile OidcCacheEntry oidcCacheEntry;

    MutableOidcCacheEntry(final OidcCacheEntry oidcCacheEntry) {
        this.oidcCacheEntry = oidcCacheEntry;
    }
}
```